### PR TITLE
Introduce inspect_level in inspector and metadata

### PR DESCRIPTION
**Pull request by MooooCat**
## Description

Inspected level is a concept newly introduced in version 0.1.5. 

Since a single column in the table may be marked by different inspectors at the same time, which may cause confuse.

Thus, a `inspect_level` is introduced when determining the specific type of a column.

## Motivation and Context

A single column, can be labeled by different multiple inspector.

For example, A email column may be recognized as email, but it may also be recognized as the id column, and it may also be recognized by different inspectors at the same time identified as a discrete column, which will cause confusion in subsequent processing. 

Also, you can see from our multi-table demo dataset, child table "train",  the `Date` column is marked twice, as discrete type and date type.

We will preset different inspector levels for different inspectors, usually more specific inspectors will get higher levels, and general inspectors (like discrete) will have inspect_level. In baseclass, the inspect_level is set to 1.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wh1isper** [comment]
I suggest limiting the priority to a range such as (1-100), the larger the priority, the sooner the execution. 

Our built-in inspector should be presented in a hierarchy (10,20,30...), so that users can insert a custom inspector from the middle.

**Comment by Guo-Yunzhe** [comment]
Nice suggestion!

I will plan it later.

從我的iPhone傳送

Zhongsheng Ji ***@***.***>於2024年1月17日 16:53寫道：

﻿

I suggest limiting the priority to a range such as (1-100), the larger the priority, the sooner the execution.

Our built-in inspector should be presented in a hierarchy (10,20,30...), so that users can insert a custom inspector from the middle.

**Review by Wh1isper — CHANGES_REQUESTED** [review]
Request some teses on `inspect_level`

File: sdgx/data_models/metadata.py [hunk 1]
```diff
@@ -44,6 +44,16 @@ class Metadata(BaseModel):
     column_list is used to store all columns' name
     """
 
+    column_inspect_level: Dict[str, int] = defaultdict()
```
Comment (Wh1isper) [inline sdgx/data_models/metadata.py:47]: Backwards compatibility does not exist for `inspect_level` as `10`.

```suggestion
    column_inspect_level: Dict[str, int] = defaultdict(lambda: 10)
```

**Review by Wh1isper — CHANGES_REQUESTED** [review]

**Comment by MooooCat** [comment]
It seems that all our package build test are failed because the SSL error of datahub.io., causing the fail of some unit test (downloading the demo dataset).

**Comment by Wh1isper** [comment]
Can a more stable data source be found? Or hitz-ids could provide a public download source similar to s3?

**Comment by Guo-Yunzhe** [comment]
Kangda Wu will work on this

從我的iPhone傳送

Zhongsheng Ji ***@***.***>於2024年1月19日 23:42寫道：

﻿

Can a more stable data source be found? Or hitz-ids could provide a public download source similar to s3?

**Review by Wh1isper — APPROVED** [review]
LGTM, just one more question, should we add some test in `test_metadata.py` for `column_inspect_level`?

**Comment by MooooCat** [comment]
> LGTM, just one more question, should we add some tests in `test_metadata.py` for `column_inspect_level`? We can leave it in next PR.

Sure, currently some test are in each inspectors, we can add test cases in `test_metadata.py`

**Pull request merged**